app/controllers/application_controller.rb

```diff
@@ -5,6 +5,130 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "1234hah"
+  end
+  
+  get '/' do
+  	erb :'/index'
+  end
+
+  get '/signup' do
+    if !!session[:user_id]
+    	redirect '/tweets'
+    else
+    	erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do
+  	if params.any?{|key, value| value.empty?}
+      redirect to '/signup'
+    else
+      @user = User.create(params)
+      session[:user_id] = @user.id
+      redirect to '/tweets'
+    end
+  end
+
+  get '/login' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'/users/login'
+    end
+  end
+
+  post '/login' do
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:id] = @user.id
+      redirect to "/tweets"
+    else
+      redirect to "/login"
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'/tweets/create_tweet'
+    else
+      redirect "/login"
+    end
+  end
+
+  post '/tweets' do
+    @tweet = Tweet.create(content: params[:content], user_id: session[:id])
+    @tweet.save
```

Comment:
Using `.create` instantiates and saves the object to the database...so there is no need to call `.save` in the following line. If you were to have done `Tweet.new`...then the `.save`. would have been necessary.